clang/include/clang/Basic/Attr.td

```diff
@@ -1082,6 +1082,13 @@ def SYCLDevice : InheritableAttr {
   let Documentation = [SYCLDeviceDocs];
 }
 
+def SYCLAllowRawPointerInKernel : Attr {
```

Comment:
So, we need an attribute to avoid diagnostics from the "standard c++ library for SYCL" implementation?
I'm not sure that it's possible to upstream such changes.

Reply:
We need this attribute to allow for explicitly marking with SYCL_EXTERNAL the functions that has a raw-pointer argument or return raw pointer, e.g. `__assert_fail`, `modf`. Any other function that does not have any raw-pointer params nor it does return raw pointer is acceptable to explicitly marking with SYCL_EXTERNAL.

Reply:
Yeah, this is I tried to say.
We implement restriction diagnosing, then violate them in unspecified extensions, then we have to workaround our diagnosing of restrictions by adding some attribute to mark functions which violate restrictions but shouldn't be diagnosed by our diagnostics.
Quite funny, right?

Reply:
Another approach implemented. Introduced a definitely knew attribute: sycl_has_definition which the same as sycl_device (SYCL_EXTERNAL). Literally, sycl_has_definition’s intent is to say ‘we have an implementation of this function somewhere in SYCL runtime or in devicelib’.